src/frontend/ui/tool/mod.rs

```diff
@@ -0,0 +1,30 @@
+use crate::frontend::{rectangle::Rectangle, screenshot_window::MouseEvent};
+
+use super::drawable::Drawable;
+
+pub mod crop;
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum ToolIdentifier {
```

Comment:
Optional suggestion:

Rename `ToolIdentifier` just to `ToolId`?

Reply:
Perhaps it's just me, but 'ToolId' evokes associations with databases and integer IDs.

Reply:
ok, then we can let it be `ToolIdentifier`.